mongo/integration/unified/event_verification.go

```diff
@@ -167,6 +167,18 @@ func verifyCommandEvents(ctx context.Context, client *clientEntity, expectedEven
 			if expected.Command != nil {
 				expectedDoc := documentToRawValue(expected.Command)
 				actualDoc := documentToRawValue(actual.Command)
+
+				// If actual.Command is empty, as is the case with redacted commands,
+				// verifyValuesMatch will return an error from DocumentOK() because
+				// there are not enough bytes to read a document from bson.RawValue{}.
+				// In the case of an empty Command, hardcode an empty bson.RawValue document.
+				if len(actual.Command) == 0 {
+					// This represents a null BSON value: first four bytes are length (7), and
+					// remaining bytes are the null value.
+					nullValue := []byte{7, 0, 0, 0, 10, 0, 0}
```

Comment:
Instead of having an empty BSON null, I think this could be simplified to the bytes representing an empty document:
```
[]byte{5, 0, 0, 0, 0}
```

Or alternatively marshal into a `bson.Raw`

```
value, _ := bson.Marshal(bson.D{})
bytes := ([]byte)(value)
```

Edit: I suspect `[]byte{7, 0, 0, 0, 10, 0, 0}` may have come from marshalling `bson.D{{}}`, which has an empty element. `[]byte{5, 0, 0, 0, 0}` is marshalled from `bson.D{}`.